dot/network/service.go

```diff
@@ -279,6 +281,18 @@ func (s *Service) logPeerCount() {
 	}
 }
 
+func (s *Service) publishNetworkTelemetry() {
```

Comment:
```suggestion
// done be passed in to shut this goroutine down if we care about shutting down gracefully
func (s *Service) publishNetworkTelemetry(done chan interface{}) {
	ticker := time.NewTicker(s.PublishNetworkTelemetryDuration)
	defer ticker.Stop()
main:
	for {
		select {
		case <-done:
			break main
		// use a channel to ensure runtime can work on other goroutines
		case <-ticker.C:
			fmt.Println("do stuff")
		}
	}
}
```

Reply:
I modified publishNetworkTelemetry as noted above, however when I added `s.doneNetworkTelemerty <- struct{}{}` to line 398 in network/service.go the node seems to hang when shutting down, and never closes.  I see this when I run TestStartService (in service_test.go).  It seems that stop gets called twice, so the channel seems to block there.  I may not have set this up correctly, so let me know if I'm missing something.